packages/modules-sdk/src/loaders/utils/load-internal.ts

```diff
@@ -0,0 +1,122 @@
+import { asFunction, asValue } from "awilix"
+import { createMedusaContainer } from "medusa-core-utils"
+import { trackInstallation } from "medusa-telemetry"
+import {
+  Constructor,
+  InternalModuleDeclaration,
+  Logger,
+  MedusaContainer,
+  ModuleExports,
+  ModuleResolution,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+} from "../../types"
+
+export async function loadInternalModule(
+  container: MedusaContainer,
+  resolution: ModuleResolution,
+  logger: Logger
+): Promise<{ error?: Error } | void> {
+  const constainerName = resolution.definition.registrationName
+
+  const { scope, resources } =
+    resolution.moduleDeclaration as InternalModuleDeclaration
+
+  let loadedModule: ModuleExports
+  try {
+    loadedModule = (await import(resolution.resolutionPath as string)).default
+  } catch (error) {
+    return { error }
+  }
+
+  if (!loadedModule?.services?.length) {
+    container.register({
+      [constainerName]: asValue(undefined),
+    })
+
+    return {
+      error: new Error(
+        "No service found in module. Make sure your module exports at least one service."
+      ),
+    }
+  }
+
+  const moduleServices = Array.isArray(loadedModule?.services)
```

Comment:
question: What made us move to allow multiple services to be exported from a module? 

The initial thinking was that we limit the entry points for module communication to a single service to make them easy to work with.

Reply:
Note, we might have discussed this in the past. If that's the case, it has slipped my mind

Reply:
I believe some modules will need multiple services, like Products.
If we were to follow the same pattern as we currently do, Inventory also would be 3 different services.